### PR TITLE
Bump dependencies: web3modal and wagmi

### PR DESCRIPTION
**Pull request by cedric25**
We currently use `@web3modal/ethereum` (https://www.npmjs.com/package/@web3modal/ethereum)

This package has been deprecated

To be used instead: https://github.com/WalletConnect/web3modal

➕ bump other related dependencies in the process.

Login avant :

Login before

Login après :

Login after

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -16,8 +18,20 @@ import {
...
 function App() {
+  const { connector } = useAccount();
+  const { chain } = useNetwork();
+
+  useEffect(() => {
+    if (!connector) {
+      return;
+    }
+    store.dispatch(initSDK({ connector }));
+  }, [connector, chain]);
```
Comment (cedric25) [inline src/App.tsx:33]: Here to watch what happens in the Metamask extension.
 - Disconnecting the wallet
 - Changing wallet
 - Changing chain

All should be supported = UI should react accordingly.

File: src/features/loginGuard/LoginGuard.tsx [hunk 2]
```diff
@@ -38,8 +34,13 @@ const LoginGuard: FC<{ children: ReactNode }> = ({ children }) => {
             Switch to {chains[0].name}
             {isLoading && pendingChainId === chains[0]?.id && ' (switching)'}
           </Button>
-          <div>{error && error.message}</div>
-        </Box>
+          {error && (
+            <div className="ml-1 mt-1.5 flex items-center text-red-500">
+              <AlertCircle size="14" />
+              <span className="ml-1 text-sm">{error.message}</span>
+            </div>
```
Comment (cedric25) [inline src/features/loginGuard/LoginGuard.tsx:41]: For example when user rejects changing chain:

Capture d’écran 2023-12-04 à 16 45 30

File: src/main.tsx [hunk 3]
```diff
@@ -30,11 +29,10 @@ root.render(
     <BrowserRouter>
       <Provider store={store}>
         <ThemeProvider theme={theme}>
-          <WagmiConfig client={wagmiClient}>
+          <WagmiConfig config={wagmiConfig}>
             <App />
             <Toaster />
           </WagmiConfig>
-          <Web3Modal projectId={projectId} ethereumClient={ethereumClient} />
```
Comment (cedric25) [inline src/main.tsx:37]: That is now done with `createWeb3Modal()` at the end of "wagmiConfig.ts" file.

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
Great work
I tested all the tricky cases I'm awaire of (account/chain switch from plugin, disconnect...) it seem's to work properly :+1: 

2 small changes to address (see comments)
- cypress ts crashing due to ts upgrade
- version in package-lock.json outdated

After the proposed cypress fix, test are still falling due to:
- the wagmi modal being different (tests to update to click the METAMAK button properly)
- the mocked metamask not being detected by wagmi (to investigate)
I suggest investigating the test cases issue in another PR

File: package-lock.json
[the remark this answers is not in the document]
Comment (cedric25) [reply package-lock.json:1]: I've added the "module" field in this specific Cypress tsconfig file 👍

**Review by PierreJeanjacquot — APPROVED** [review]
LGTM, tests to fix in a next PR

**Pull request merged**